Lesson1 (Sequence containers)/library.cpp

```diff
@@ -0,0 +1,50 @@
+#include <iostream>
+#include "library.h"
+
+
+
+std::vector<int> combineVectors (const std::vector<int>::iterator& begin1, const std::vector<int>::iterator& end1, const std::vector<int>::iterator& begin2, const std::vector<int>::iterator& end2)
+{
+    ulong size = (end1 - begin1) + (end2 - begin2);
+    std::vector<int> tmp;
+    tmp.reserve(size);
+    tmp.insert(tmp.begin(), begin1, end1);
+    tmp.insert(tmp.end(),begin2, end2);
+
+    return tmp;
+};
+
+size_t magicNumber(const size_t number, const size_t degree)
+{
+    long wantedNumber = 0;
+    std::list<long> numbers;
+    numbers.push_back(number);
+
+    for (auto&& it = numbers.begin(); it != numbers.end(); ++it) {
+        if (numbers.size() < degree)
+            numbers.push_back(*(--numbers.end()) * *(numbers.begin()));
+
+        long first = *it;
+        long last = (*it) % 10;
```

Comment:
**hm, the list looks absolutely useless here.**

19^1 = **1****9**
19^2 = **3**6**1**
19^3 = **6**85**9**
19^4 = **1**3032**1**

You should work with the first and the last element in list, without usage of ariphmetic operations.
Alternatively, you can use the string representation.